File: RWfile.cpp
/* 
 *  example of using iostream and fstream libraries for c++ file io
 *  
 *  Al Curry     February, 2017
 * 
 * For this example, input file is assumed to be in the same directory from which the compiled
 * program is run.  The output file will be placed here as well.   
 * 
 * A slicker option would allow the filename and io directories to be specified as program arguments, 
 * passed them in on invocation, or prompted for.  But again, this program could be replaced by your
 * OS copy command, it is merely an extecise.
 *  
 */

#include <iostream>
#include <fstream>
#include <cstring>

using namespace std;

int main() {

	string line;
	ofstream outputFile;
	ifstream inputFile;

	inputFile.open("ac.txt");
	outputFile.open("Curry.txt");  // options can be added to open rather than overwrite 
                                       // existing file, also for binary rather than text mode

	if (inputFile.is_open()) {
		while (getline(inputFile, line)) {

			//cout << "number of characters in line below: " << line.length() << "\n";
			//cout << line << "\n";
                    
                        // send the line read in to the outputFile stream, followed by a newline character 
			outputFile << line << "\n";

		}
	} else {
		cout << "file not opened\n";
	}

	outputFile.close();
	inputFile.close();

	return 0;
}

File: RWfile.py
#
#  File i/o with Python
#
#  Al Curry   March 2016
#
#  ac.txt assumed to be in the same directory as this file RWfile.py
#
#

# open file to read in read mode, "r"
read_file = open("ac.txt","r")

# open file to be written to in write mode, "w"
write_file = open("Curry.txt","w")

for line in read_file:

    # strip function removes the line feed when reading in, otherwise
    # "print" would create double-spaced output, since it adds a line feed
    # write the line out with the line feed added back
    line = line.strip()
    print(line)
    write_file.write(line + "\n")

# close the files upon completion
read_file.close()
write_file.close()
